### PR TITLE
Allow OpenMP transformations with Intergrid kernels

### PR DESCRIPTION
**Pull request by arporter**

### CODE REVIEW

**Comment by arporter** [comment]
Hi @christophermaynard, this is the promised PR. Please could you try building your version of LFRic with OpenMP enabled and check that all is well?

**Comment by arporter** [comment]
Note, I've been naughty and also included a fix to get the documentation build working on readthedocs (the fix is a pip requirements file that lists what needs to be installed in order to build the documentation).

**Comment by TeranIvy** [comment]
I tested this functionality on Tom's LFRic branch which adds multigrid solver in the gravity wave miniapp [(LFRic ticket 1547)](https://code.metoffice.gov.uk/trac/lfric/ticket/1547). He excluded two multigrid invokes (`call invoke( name = "Multigrid_Restriction", ...` and `call invoke( name = "Multigrid_Prolongation", ...` ) in `gw_multigrid_precon_alg_mod.x90` from optimisations as this is not enabled in current 1.6.0 release.

I built PSyclone from this branch and enabled optimisations from `global.py` (colouring and OpenMP) for these two invokes as well. The original LFRic branch was built with PSyclone 1.6.0. The generated code is correct (comparison with the original LFRic branch where there is no colouring and OpenMP for these two invokes) and the executable runs fine.
The outputs were compared with the original LFRic branch for 1, 2 and 4 threads (both serial and parallel runs). The results are the same between the modified and the original branch as well as between threads. 

All is well - please pass to Rupert for code review so he doesn't become lazy :grin:

**Comment by rupertford** [comment]
@TeranIvy has kindly picked this up and checked that OpenMP works with intergrid kernels. I will now review.

File: src/psyclone/transformations.py [hunk 1]
```diff
@@ -928,9 +928,6 @@ def apply(self, node):
         :py:class:`base class <OMPParallelLoopTrans>`. '''
         OMPParallelLoopTrans._validate(self, node)
 
-        # Check that we don't have an inter-grid kernel
```
Comment (rupertford) [inline src/psyclone/transformations.py]: Documentation of the associated method needs to define the interface.

Comment (rupertford) [reply src/psyclone/transformations.py]: Done.

**Pull request merged**